lang/ruby/ruby_find_pkgsdeps

```diff
@@ -71,111 +89,260 @@ require_ignore=%w{
+			when "require"
+				#[:command, [:@ident, "require", [3, 0]], [:args_add_block, [[:string_literal, [:string_content, [:@tstring_content, "pathname", [3, 9]]]]], false]]
+				#[:command, [:@ident, "require", [3, 0]], [:args_add_block, [[:string_literal, [:string_content, [:@tstring_content, "rbconfig", [3, 9]]]]], false]]
+				#[:command, [:@ident, "require", [3, 0]], [:args_add_block, [[:string_literal, [:string_content, [:@tstring_content, "rubygems/dependency", [3, 9]]]]], false]]
+				# Only accepts requires with only a literal strings (without embeded expressions)
+				req = node[2][1][0][1][1][1] if node[2][1][0][1][1][0] == :@tstring_content and node[2][1][0][1].length == 2 rescue nil
+				requires << req if req
+			end
 
-puts "Looking for requires in files..."
-files_requires=Hash.new { |h,k| h[k]=[] }
-packages.each do
-        |pkg|
-	package_files[pkg].each do
-		|file|
-		next if not File.file?(file)
+			# node = [:command, [:@ident, "pp", [ln,col]], [:args_add_block, ...]]
+			if node[1][0] == :@ident && node[1][1] == "pp"
+				requires << "pp"
+			end
 
-		if not file =~ /.rb$/
-			if File.executable?(file)
-				magic=`head -c50 '#{file}' | head -1`
-				begin
-					if not magic =~ /ruby/
-						next
+		when :command_call
+			req = nil
+			# node = [:command_call, receiver, [:@period, ".", [ln,col]], [:@ident, "require", [ln,col]], args]
+			if node[3][1] == "require"
+					# Only accepts requires with only a literal strings (without embedded expressions)
+					args = node[4] rescue nil
+					if args && args[1] && args[1][0] && args[1][0][1] && args[1][0][1][1][0] == :@tstring_content && args[1][0][1].length == 2
+							req = args[1][0][1][1][1] rescue nil
+							requires << req if req
 					end
-				rescue
-					next
-				end
-			else
-				next
+			end			
+		# The args in Encoding.find (if a string constant) should also requires the encoding (but,in practice,
+		# there is no case for it current ruby code
+		when :const_path_ref
+			# [:const_path_ref, [:var_ref, [:@const, "Encoding", [136, 9]]], [:@const, "US_ASCII", [136, 19]]]
+			if node[1][0]=:const_ref and node[1][1][1] == "Encoding" and node[2][0] = :@const
+				enc = node[2][1]
+
+				enc = eval("Encoding::#{enc}")
+				encodings << enc if enc.kind_of? Encoding
+
+				# The builtin encodings do not populate Encoding::XXX constants
+				requires << "enc/encdb"
+			end
+
+		    	node.each do |child|
+				stack << child if child.is_a?(Array)
+			end
+
+		when :method_add_arg
+			# Detects fcall :pp => [:method_add_arg, [:fcall, [:@ident, "pp", [1,0]]], ...]
+			if node[1][0] == :fcall && node[1][1][0] == :@ident && node[1][1][1] == "pp"
+				requires << "pp"
+			end
+
+			# Detects Kernel.pp => [:method_add_arg, [:call, [:var_ref, [:@const, "Kernel", [1,0]]], :".", [:@ident, "pp", [1,8]]], ...]
+			if node[1][0] == :call &&
+			   node[1][1][0] == :var_ref && node[1][1][1][1] == "Kernel" &&
+		           node[1][3][0] == :@ident && node[1][3][1] == "pp"
```

Comment:
Again mixing tabs and spaces and thing don't seem to align with 4-space tabs.

<img width="302" height="94" alt="image" src="https://github.com/user-attachments/assets/e3278d46-3898-47db-a632-1970b8092266" />

Reply:
Here again. Fixed. Thanks